spot-contracts/contracts/BondIssuer.sol

```diff
@@ -113,7 +124,35 @@ contract BondIssuer is IBondIssuer, OwnableUpgradeable {
 
     /// @inheritdoc IBondIssuer
     function isInstance(IBondController bond) external view override returns (bool) {
-        return _issuedBonds.contains(address(bond));
+        return (_activeBonds.contains(address(bond)) || _matureBonds.contains(address(bond)));
+    }
+
+    /// @inheritdoc IBondIssuer
+    /// @dev Reverts if none of the active bonds are mature.
+    function matureAll() external {
+        bool bondsMature = false;
+        uint256 activeCount_ = _activeBonds.length();
+
+        // NOTE: We traverse the active list in the reverse order as deletions involve
+        //       swapping the deleted element to the end of the list and removing the last element.
+        for (uint256 i = activeCount_; i > 0; i--) {
+            IBondController bond = IBondController(_activeBonds.at(i - 1));
+            if (bond.timeToMaturity() == 0) {
+                if (!bond.isMature()) {
+                    bond.mature();
+                }
+
+                _activeBonds.remove(address(bond));
+                _matureBonds.add(address(bond));
+                emit BondMature(bond);
+
+                bondsMature = true;
+            }
+        }
+
+        if (!bondsMature) {
```

Comment:
Is there a benefit to having this revert? It seems like we're using an error reporting feature for returning data to the caller.

In the case of no maturity events, I'd expect it to be a no-op, return a bool, or return an integer representing the number of bonds that matured (0).

Reply:
The reverts are useful when we hook this up to multiple keeper services like gelato, chainlink  and probably our own. The first successful call succeeds and the others fail, so the off-chain keeper script doesn't execute.

Reply:
More generally I'm currently leaning toward poke-able functions (like rebase) reverting. That way off-chain callers can be dumb and just keep polling until its executable and trigger it.